src/Microsoft.SqlTools.ServiceLayer/LanguageExtensibility/ExternalLanguageOperations.cs

```diff
@@ -0,0 +1,359 @@
+            string contentAction = "FROM";
+            if (modifyType == ModifyType.Alter)
+            {
+                switch (contentModifyType)
+                {
+                    case ContentModifyType.Modify:
+                        contentAction = SetContentScript;
+                        break;
+                    case ContentModifyType.Add:
+                        contentAction = AddContentScript;
+                        break;
+                    case ContentModifyType.Remove:
+                        contentAction = RemoveContentScript;
+                        break;
+                }
+            }
+            return $@"
+{scriptAction} {language.Name}
+{ownerScript}
+{contentAction} {contentScript}
+";
+        }
+
+        private string AddStringParameter(string paramName, string prefix, string postfix, string paramValue, Dictionary<string, object> parameters)
+        {
+            string script = string.Empty;
+            string sqlParam = $"@{paramName}{postfix}";
+            /*
+            if (!string.IsNullOrWhiteSpace(paramValue))
+            {
+                script = $", {paramName} = N'{sqlParam}'";
+                parameters.Add($"{sqlParam}", paramValue);
+            }
+            */
+
+            if (!string.IsNullOrWhiteSpace(paramValue))
+            {
+                script = $"{prefix} {paramName} = N'{paramValue}'";
+            }
+            return script;
+        }
+
+        private string GetLanguageContent(ExternalLanguageContent content, int index, Dictionary<string, object> parameters)
+        {
+            string postfix = index.ToString();
+            string prefix = ",";
+            string contentScript = string.Empty;
+            if (content.IsLocalFile && content.Content != null)
+            {
+                parameters.Add($"{ContentParamName}{postfix}", content.Content);
+                contentScript = $"CONTENT = @{ContentParamName}{postfix}";
+            }
+            else if (!string.IsNullOrWhiteSpace(content.PathToExtension))
+            {
+                contentScript = $"{AddStringParameter(ContentParamName, string.Empty, postfix, content.PathToExtension, parameters)}";
+            }
+            return $@"( 
+                      {contentScript}
+                      {AddStringParameter(FileNameParamName, string.IsNullOrWhiteSpace(contentScript) ? string.Empty : prefix,
+                      postfix, content.ExtensionFileName, parameters)}
+                      {AddStringParameter(PlatformParamName, prefix, postfix, content.Platform, parameters)}
+                      {AddStringParameter(ParametersParamName, prefix, postfix, content.Parameters, parameters)}
+                      {AddStringParameter(EnvVariablesParamName, prefix, postfix, content.EnvironmentVariables, parameters)}
+                      )";
+        }
+
+        private void ExecuteNonQuery(IDbConnection connection, string script, Dictionary<string, object> parameters)
```

Comment:
This is super generic - do we not have something like this in a general utils class already?

Reply:
the code will change to SMO hopefully soon so this function wont' be used that much. most of the services are using SMO already 

---
In reply to: [374373722](https://github.com/microsoft/sqltoolsservice/pull/918#discussion_r374373722) [](ancestors = 374373722)